repo-agent/pkg/codeserver/codeserver.go

```diff
@@ -0,0 +1,31 @@
+package codeserver
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var execCommand = exec.Command
+
+func Start() (*exec.Cmd, error) {
+	log.Println("starting code-server")
+	repoURL := os.Getenv("GIT_HTML_URL")
+	parts := strings.Split(strings.TrimPrefix(repoURL, "https://github.com/"), "/")
```

Comment:
Parsing the repository name from a URL with `strings.Split` is brittle. This will fail for URLs that don't match the exact `https://github.com/a/b/c/d` format. Consider using the `net/url` package for more robust URL parsing.